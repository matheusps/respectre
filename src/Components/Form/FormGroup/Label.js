import React from 'react'

const Label = props => (
    <label {...props}></label>
)

export default Label