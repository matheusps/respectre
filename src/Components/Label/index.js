import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const Label = ({ className, theme, rounded, children }) => {
  const labelClaasses = classNames(className, 'label', `label-${theme}`, {
    'label-rounded': rounded,
  })
  return <span className={labelClaasses}>{children}</span>
}

Label.propTypes = {
  /** Aditional css classes */
  className: PropTypes.string,
  /** Label theme */
  theme: PropTypes.oneOf([
    'primary',
    'secondary',
    'success',
    'warning',
    'error',
  ]),
  /** If has rounded corners or not */
  rounded: PropTypes.bool,
}

export default Label
