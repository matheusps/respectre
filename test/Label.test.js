import React from 'react'
import Label from '../src/Components/Label'
import renderer from 'react-test-renderer'

describe('Label', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Label />).toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should match snapshot with props', () => {
    const component = renderer
      .create(
        <Label className="testing" theme="primary" style={{ float: 'left' }} />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  it('should have the correct props', () => {
    const component = renderer
      .create(
        <Label
          className="testing"
          theme="primary"
          rounded
          style={{ float: 'left' }}
        />
      )
      .toJSON()

    expect(component.props.className).toContain('testing')
    expect(component.props.className).toContain('label')
    expect(component.props.className).toContain('label-primary')
    expect(component.props.className).toContain('label-rounded')
    expect(component.props.style).toEqual({ float: 'left' })
  })
})
