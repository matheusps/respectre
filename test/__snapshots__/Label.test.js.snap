// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should match snapshot 1`] = `
<span
  className="label label-undefined"
/>
`;

exports[`Label should match snapshot with props 1`] = `
<span
  className="testing label label-primary"
  style={
    Object {
      "float": "left",
    }
  }
/>
`;
